### ISSUE TITLE
Imported transaction with 0 as id

### ISSUE THREAD

**Issue by nizarus** [firefly-iii/firefly-iii#9478]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [x] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

A transaction imported from a CSV file with "Firefly III Data Importer, v1.5.7" got a 0 as id `somedomain.com/firefly/public/transactions/show/0` so it can't be edited and or deleted.

### Debug information

Debug information generated at 2024-11-18 15:52:55 for Firefly III version **v6.1.22**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.22 / v2.1.0 / 25 (exp. 25)</td></tr><tr><td>PHP version</td><td>8.3.13 (64bits) / litespeed / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>2097152 (2 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Africa/Tunis + Africa/Tunis</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>fr_FR + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-07-01 13:13:39</td></tr><tr><td>Session end</td><td>2024-07-31 13:13:39</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>fr_FR</td></tr><tr><td>User locale</td><td>fr_FR</td></tr><tr><td>Locale(s) supported</td><td>fr_FR.utf8: :white_check_mark:<br>fr_FR.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0</td></tr></tbody></table>

**Comment by JC5** [comment]
Hi, thanks for opening an issue.

I'm sorry, I need a bit more detail. I assume that somewhere you found a link that goes to transaction nr. 0? Could you be a little bit more specific?

I don't believe for a second MySQL would hand out id nr. 0.

**Comment by nizarus** [comment]
Hi,
When visiting the withdrawal page : `/firefly/public/transactions/withdrawal/` I found a transaction without a category :
![Image](https://github.com/user-attachments/assets/b8c4455e-31a5-47ab-a2ff-7cb684a73f0d)
When I try to display it or to delete it, via actions, button, I'm redirected to a "404 not found" page as the link is terminated by a 0 `/transactions/edit/0` or `/transactions/show/0`. I'm calling 0 the id, but I can be wrong :-)
For information, this transaction with id 0 is between two other transaction with ids 700 and 701.

**Comment by JC5** [comment]
Alright, that helps. Can you reproduce the issue when you import it again?

**Comment by nizarus** [comment]
No, I imported the same transaction a second time, and it was successfully imported with a non 0 id.
Actually, the same transaction is displayed and counted twice.
